hazelcast-client-new/src/test/java/com/hazelcast/client/quorum/ClientTransactionalMapQuorumTest.java

```diff
@@ -0,0 +1,379 @@
+            fail();
+        } catch (TransactionException e) {
+            transactionContext.rollbackTransaction();
+        }
+    }
+
+    @Test
+    public void testTxContainsKeyThrowsExceptionWhenQuorumSizeNotMet() {
+        TransactionContext transactionContext = c4.newTransactionContext(options);
+        try {
+            transactionContext.beginTransaction();
+            TransactionalMap<Object, Object> map = transactionContext.getMap(randomMapName(MAP_NAME_PREFIX));
+            map.containsKey("foo");
+            transactionContext.commitTransaction();
+            fail();
+        } catch (TransactionException e) {
+            transactionContext.rollbackTransaction();
+        }
+    }
+
+
+    @Test
+    public void testTxIsEmptyThrowsExceptionWhenQuorumSizeNotMet() {
+        TransactionContext transactionContext = c4.newTransactionContext(options);
+        try {
+            transactionContext.beginTransaction();
+            TransactionalMap<Object, Object> map = transactionContext.getMap(randomMapName(MAP_NAME_PREFIX));
+            map.isEmpty();
+            transactionContext.commitTransaction();
+            fail();
+        } catch (TransactionException e) {
+            transactionContext.rollbackTransaction();
+        }
+    }
+
+    @Test
+    public void testTxKeySetThrowsExceptionWhenQuorumSizeNotMet() {
+        TransactionContext transactionContext = c4.newTransactionContext(options);
+        try {
+            transactionContext.beginTransaction();
+            TransactionalMap<Object, Object> map = transactionContext.getMap(randomMapName(MAP_NAME_PREFIX));
+            map.keySet();
+            transactionContext.commitTransaction();
+            fail();
+        } catch (TransactionException e) {
+            transactionContext.rollbackTransaction();
+        }
+    }
+
+    @Test
+    public void testTxKeySetWithPredicateThrowsExceptionWhenQuorumSizeNotMet() {
+        TransactionContext transactionContext = c4.newTransactionContext(options);
+        try {
+            transactionContext.beginTransaction();
+            TransactionalMap<Object, Object> map = transactionContext.getMap(randomMapName(MAP_NAME_PREFIX));
+            map.keySet(TruePredicate.INSTANCE);
+            transactionContext.commitTransaction();
+            fail();
+        } catch (TransactionException e) {
+            transactionContext.rollbackTransaction();
+        }
+    }
+
+    @Test
+    public void testTxValuesThrowsExceptionWhenQuorumSizeNotMet() {
+        TransactionContext transactionContext = c4.newTransactionContext(options);
+        try {
+            transactionContext.beginTransaction();
+            TransactionalMap<Object, Object> map = transactionContext.getMap(randomMapName(MAP_NAME_PREFIX));
+            map.values();
+            transactionContext.commitTransaction();
+            fail();
+        } catch (TransactionException e) {
+            transactionContext.rollbackTransaction();
```

Comment:
You might be right, as far as I remember transaction context is not re-usable. But I'm not %100 sure about that.